pandas/core/base.py

```diff
@@ -517,9 +517,16 @@ def _make_str_accessor(self):
             raise AttributeError("Can only use .str accessor with string "
                                  "values, which use np.object_ dtype in "
                                  "pandas")
-        elif isinstance(self, Index) and self.inferred_type != 'string':
-            raise AttributeError("Can only use .str accessor with string "
-                                 "values (i.e. inferred_type is 'string')")
+        elif isinstance(self, Index):
+            # see scc/inferrence.pyx which can contain string values
+            allowed_types = ('string', 'unicode', 'mixed', 'mixed-integer')
+            if self.inferred_type not in allowed_types:
+                message = ("Can only use .str accessor with string values "
+                           "(i.e. inferred_type is 'string', 'unicode' or 'mixed')")
```

Comment:
@jreback It is to be compat with current ``Series`` behavior. In above case, `Series.str` is applied to all the elements and leave non-str as ``NaN``.

```
s = pd.Series([1, 2, 'a'])
s.str.len()
# 0   NaN
# 1   NaN
# 2     1
# dtype: float64
```